sotodlib/mapmaking/demod_mapmaker.py

```diff
@@ -0,0 +1,308 @@
+import numpy as np
+from pixell import enmap, utils, tilemap, bunch
+import so3g.proj
+
+from .. import coords
+from .utilities import recentering_to_quat_lonlat, evaluate_recentering
+
+class DemodMapmaker:
+    def __init__(self, signals=[], noise_model=None, dtype=np.float32, verbose=False, comps='TQU', singlestream=False):
+        """
+        Initialize a FilterBin Mapmaker for demodulated data 
+        
+        Arguments
+        ---------
+        signals : list 
+            List of Signal-objects representing the models that will be solved
+            jointly for. Currently this would be a DemodSignal
+        noise_model : sotodlib.mapmaking.Nmat or None
```

Comment:
Hi, Carlos. I am jumping into this PR.
Can you add more docstrings here and that of add_obs for the noise_model?
It does not make sense to me what happens when you leave this None and how you can arrange it.

Reply:
For example, I guess that the most frequent case people want to do is get variance from fitting of spectrum and use it as weight for map making. How people can do that? And is the default supposed to so? 

When I refer to the Nmat, it looks calculating the variance from just rms of tods. That maybe fine, but I am not so confident. We should make sure at least to make people aware of what types weights they are using.

Reply:
ok, I'm adding more to the docstring. Yes, the NmatWhite model calculates the variance from the timestream as rms. We can implement more stuff in the Nmat classes by creating new noise models.